axelor-admin/src/main/java/com/axelor/apps/base/service/app/DataBackupCreateService.java

```diff
@@ -175,31 +179,87 @@ public File create(Integer fetchLimit) {
+  private Map<String, List<String>> getSubClassesMap() {
+    List<MetaModel> metaModels = getMetaModels();
+    Map<String, List<String>> subClassMap = new HashMap<String, List<String>>();
+    List<String> subClasses;
+    Class<Model> superClass, subClass;
+    for (MetaModel metaModel : metaModels) {
+      subClasses = new ArrayList<>();
+      try {
+        superClass = (Class<Model>) Class.forName(metaModel.getFullName());
+        for (MetaModel metaModelItem : metaModels) {
+          subClass = (Class<Model>) Class.forName(metaModelItem.getFullName());
+          if (subClass.getSuperclass() == superClass) {
+            subClasses.add(metaModelItem.getName());
+          }
+        }
+      } catch (Exception e) {
+      }
+      if (subClasses != null && subClasses.size() > 0) {
+        subClassMap.put(metaModel.getFullName(), subClasses);
+      }
+    }
+    return subClassMap;
+  }
+
   /* Get All Data of Specific MetaModel */
-  private List<Model> getMetaModelDataList(MetaModel metaModel, int start, Integer fetchLimit) {
+  private List<Model> getMetaModelDataList(
+      MetaModel metaModel, int start, Integer fetchLimit, List<String> subClasses)
+      throws ClassNotFoundException {
+    String query = "";
+    if (subClasses != null && subClasses.size() > 0) {
+      String whereStr = "";
+      for (String subClassName : subClasses) {
+        whereStr += "id NOT IN (select id from " + subClassName + ")";
+        if (whereStr.length() > 0 && !subClassName.equals(subClasses.get(subClasses.size() - 1))) {
+          whereStr += " AND ";
+        }
+      }
+      query = "FROM " + metaModel.getName() + " WHERE " + whereStr;
+    } else {
+      query = "FROM " + metaModel.getName();
+    }
     return JPA.em()
-        .createQuery("FROM " + metaModel.getName(), Model.class)
+        .createQuery(query, Model.class)
         .setHint(QueryHints.MAINTAIN_CACHE, HintValues.FALSE)
         .setFirstResult(start)
         .setMaxResults(fetchLimit)
         .getResultList();
   }
 
-  private long getMetaModelDataCount(MetaModel metaModel)
+  private long getMetaModelDataCount(MetaModel metaModel, List<String> subClasses)
       throws InterruptedException, ClassNotFoundException {
-    return (long)
-        JPA.em().createQuery("SELECT count(*) FROM " + metaModel.getName()).getSingleResult();
+    String query = "";
+    if (subClasses != null && subClasses.size() > 0) {
+      String whereStr = "";
+      for (String subClassName : subClasses) {
+        whereStr += "id NOT IN (select id from " + subClassName + ")";
+        if (whereStr.length() > 0 && !subClassName.equals(subClasses.get(subClasses.size() - 1))) {
+          whereStr += " AND ";
+        }
+      }
+      query = "SELECT count(*) FROM " + metaModel.getName() + " WHERE " + whereStr;
+    } else {
+      query = "SELECT count(*) FROM " + metaModel.getName();
+    }
+    return (long) JPA.em().createQuery(query).getSingleResult();
```

Comment:
Method for count and getting model should be only one. There is no need of count(*) funciton, but you can use count() method of query.  Use JpaRepository.of(modelClass) instead of JPA.em()